### PR TITLE
Update index.md

### PR DESCRIPTION
**Pull request by 0xObsidian**
Included Reth in the execution client table, propose change.

### CODE REVIEW

File: src/content/developers/docs/nodes-and-clients/index.md [hunk 1]
```diff
@@ -139,6 +139,7 @@ This table summarizes the different clients. All of them pass [client tests](htt
 | [Nethermind](http://nethermind.io/)             | C#, .NET | Linux, Windows, macOS | Mainnet, Sepolia, Goerli, and more | Snap (without serving), Fast, Full | Archive, Pruned |
 | [Besu](https://besu.hyperledger.org/en/stable/) | Java     | Linux, Windows, macOS | Mainnet, Sepolia, Goerli, and more | Snap, Fast, Full                   | Archive, Pruned |
 | [Erigon](https://github.com/ledgerwatch/erigon) | Go       | Linux, Windows, macOS | Mainnet, Sepolia, Goerli, and more | Full                               | Archive, Pruned |
+| [Reth](https://github.com/paradigmxyz/reth)     | Rust     | Linux, Windows, macOS | Mainnet, Sepolia, Goerli, and more | Full                               | Archive, Pruned |
```
Comment (RamanSB) [inline src/content/developers/docs/nodes-and-clients/index.md:142]:
```suggestion
| [Reth](https://github.com/paradigmxyz/reth)     | Rust     | Linux, Windows, macOS | Mainnet, Sepolia, Goerli, and more | Snap, Fast, Full                   | Archive, Pruned |
```

**Review by RamanSB — APPROVED** [review]

File: src/content/developers/docs/nodes-and-clients/index.md (hunk 1 above)
Comment (RamanSB) [inline src/content/developers/docs/nodes-and-clients/index.md:142]:
```suggestion
| [Reth](https://github.com/paradigmxyz/reth)     | Rust     | Linux, Windows, macOS | Mainnet, Sepolia, Goerli, and more | Snap, Fast, Full                   | Archive, Pruned |
```

**Review by RamanSB — CHANGES_REQUESTED** [review]

File: src/content/developers/docs/nodes-and-clients/index.md [hunk 2]
```diff
@@ -139,6 +139,7 @@ This table summarizes the different clients. All of them pass [client tests](htt
 | [Nethermind](http://nethermind.io/)             | C#, .NET | Linux, Windows, macOS | Mainnet, Sepolia, Goerli, and more | Snap (without serving), Fast, Full | Archive, Pruned |
 | [Besu](https://besu.hyperledger.org/en/stable/) | Java     | Linux, Windows, macOS | Mainnet, Sepolia, Goerli, and more | Snap, Fast, Full                   | Archive, Pruned |
 | [Erigon](https://github.com/ledgerwatch/erigon) | Go       | Linux, Windows, macOS | Mainnet, Sepolia, Goerli, and more | Full                               | Archive, Pruned |
+| [Reth](https://github.com/paradigmxyz/reth)     | Rust     | Linux, Windows, macOS | Mainnet, Sepolia, Goerli, and more | Snap, Fast, Full                   | Archive, Pruned |
```
Comment (corwintines) [inline src/content/developers/docs/nodes-and-clients/index.md:142]:
```suggestion
| [Reth](https://github.com/paradigmxyz/reth)     | Rust     | Linux, Windows, macOS | Mainnet, Sepolia, Goerli, and more | Full                   | Archive, Pruned |
```

Couldn't find documentation on these other syncing strategies, so going to revert this back to just full

**Review by corwintines — APPROVED** [review]
Thanks @0xObsidian

**Comment by corwintines** [comment]
@all-contributors can you add @0xObsidian for content

**Pull request merged**